src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/Runners/MonitorRunner.cs

```diff
@@ -32,7 +32,9 @@ internal class MonitorRunner : IAsyncDisposable
 
         private readonly LoggingRunnerAdapter _adapter;
 
-        private bool _isDisposed;
+        private readonly TemporaryDirectory _tempDir;
+
+        private long _disposedState;
```

Comment:
Consider something like AsyncDisposable that has DisposeAsync and DisposeAsyncCore. The non-virtual DisposeAync can do checks against the state flag and DisposeAsyncCore can be virtual.

Reply:
This might be a good candidate for a [source generator](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) instead of creating a base class. I'm going to leave these changes as-is and log an issue to consider each solution.